.github/workflows/publish.yml

```
@@ -0,0 +1,67 @@
+name: Publish
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  publish:
+    runs-on: ubuntu-16.04
+
+    env:
+      GRADLE_ARGS: >-
+        --no-daemon
+        --max-workers 2
+        -Pkotlin.incremental=false
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Gradle cache
+        uses: actions/cache@v1
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+            ~/.konan
+          key: ${{ runner.os }}-publish-${{ hashFiles('**/build.gradle.kts') }}
+          restore-keys: |
+            ${{ runner.os }}-publish-
+            ${{ runner.os }}-
+
+      - name: Check
+        run: ./gradlew $GRADLE_ARGS check
+
+      - name: Keyring
+        run: echo "${{ secrets.SIGNING_SECRET_KEY_RING }}" | base64 --decode > ~/secring.gpg
+
+      - name: Maven Publish
```

nit: Since you'll only have one kind of publish, could shorten this.

```suggestion
      - name: Publish
```